tests/test_backend.py

```diff
@@ -0,0 +1,422 @@
+        )
+
+        with MockServer(DelayedResource) as server:
+
+            @attr.define
+            class ItemPage(ItemWebPage):
+                http_client: HttpClient
+
+                async def to_item(self):
+                    try:
+                        await self.http_client.request(
+                            server.root_url,
+                            body=b"0.002",
+                        )
+                    except HttpRequestError:
+                        return {'foo': 'bar'}
+
+            class TestSpider(Spider):
+                name = 'test_spider'
+                start_urls = [server.root_url]
+
+                custom_settings = {
+                    'DOWNLOADER_MIDDLEWARES': {
+                        'scrapy_poet.InjectionMiddleware': 543,
+                    },
+                }
+
+                async def parse(self, response, page: ItemPage):
+                    item = await page.to_item()
+                    items.append(item)
+
+            crawler = make_crawler(TestSpider, {})
+            yield crawler.crawl()
+
+    assert items == [{'foo': 'bar'}]
+
+
+@inlineCallbacks
+def test_additional_requests_ignored_request():
+    items = []
+
+    with MockServer(EchoResource) as server:
+
+        @attr.define
+        class ItemPage(ItemWebPage):
+            http_client: HttpClient
+
+            async def to_item(self):
+                try:
+                    await self.http_client.request(
+                        server.root_url,
+                        body=b'ignore',
+                    )
+                except HttpError as e:
+                    return {'exc': e.__class__}
+
+        class TestDownloaderMiddleware:
+            def process_response(self, request, response, spider):
+                if b'ignore' in response.body:
+                    raise IgnoreRequest
+                return response
+
+        class TestSpider(Spider):
+            name = 'test_spider'
+            start_urls = [server.root_url]
+
+            custom_settings = {
+                'DOWNLOADER_MIDDLEWARES': {
+                    TestDownloaderMiddleware: 1,
+                    'scrapy_poet.InjectionMiddleware': 543,
+                },
+            }
+
+            async def parse(self, response, page: ItemPage):
+                item = await page.to_item()
+                items.append(item)
+
+        crawler = make_crawler(TestSpider, {})
+        yield crawler.crawl()
+
+    assert items == [{'exc': HttpError}]
+
+
+@pytest.mark.xfail(
+    reason=(
+        "Currently, we do not make a distinction between exceptions raised "
+        "from the downloader or from a downloader middleware, except for "
+        "IgnoreRequest. In the future, we might want to inspect the stack to "
+        "determine the source of an exception and raise HttpError instead of "
+        "HttpRequestError when the exception originates in a downloader "
+        "middleware."
+    ),
+    strict=True,
+)
```

Comment:
Is it OK to leave this out of the scope of this pull request, or should I look into addressing this here?

Reply:
I think that's ok.